packages/xo-web/src/common/xo/index.js

```diff
@@ -1806,7 +1806,32 @@ export const editVm = async (vm, props) => {
   }
   await _call('vm.set', { ...props, id: resolveId(vm) })
     .catch(err => {
-      error(_('setVmFailed', { vm: renderXoItemFromId(resolveId(vm)) }), err.message)
+      // error code 13 is vmBadPowerState "VM state is xxx but should be halted"
+      if ((err.code === 13 && err.data.actual === 'running') || err.message === 'Cannot change memory on running VM') {
```

Comment:
You can import the error `vmBadPowerState` from `api-errors.js` and then test `vmBadPowerState.is(err)` to make it more explicit.